datalad/support/gitrepo.py

```diff
@@ -3627,3 +3639,33 @@ def _fixup_submodule_dotgit_setup(ds, relativepath):
     lgr.debug(
         'Not retro-fitting GitRepo with deprecated symbols, '
         'datalad-deprecated package not found')
+
+
+def _check_for_win_compat(files):
+    """Check file names for illegal characters or reserved names on Windows
+    
+    In the case that a non-Windows-compatible file is detected, warn users
+    about potential interoperability issues.
+    """
+    for file in files.keys():
+        f = Path(file).stem
```

Comment:
I wonder whether this should cover directories too? The could also have illegal chars.

Reply:
oh you're right, I thought it would only be files, but indeed it applies to directories as well

Reply:
done